python/pyarrow/tests/test_dataset.py

```diff
@@ -589,6 +589,7 @@ def test_partitioning():
         partitioning = klass(schema)
         assert isinstance(partitioning, ds.Partitioning)
         assert partitioning == klass(schema)
+        assert partitioning != "other object"
```

Comment:
I am not sure I understand why the new assert is necessary here but I am happy with the fix

Reply:
It was this kind of assert that segfaulted, so adding it here to verify that doesn't happen (I am not fully sure why dask was doing something like this, though)

Reply:
ok, sounds good